test/src/lib/update-lock.spec.ts

```diff
@@ -0,0 +1,273 @@
+import { expect } from 'chai';
+import { SinonSpy, SinonStub, spy, stub } from 'sinon';
+import * as path from 'path';
+import * as Bluebird from 'bluebird';
+
+import rewire = require('rewire');
+import mockFs = require('mock-fs');
+
+import * as constants from '../../../src/lib/constants';
+import { UpdatesLockedError } from '../../../src/lib/errors';
+
+describe('lib/update-lock', () => {
+	const updateLock = rewire('../../../src/lib/update-lock');
```

Comment:
glad to see rewire being useful for others as well. I've really enjoyed it.

Reply:
Same! It's been invaluable in tests.